model/host/db.go

```diff
@@ -937,16 +955,27 @@ func RemoveStaleInitializing(distroID string) error {
 	if err := db.FindAllQ(Collection, q, &hosts); err != nil {
 		return errors.WithStack(err)
 	}
-	ids := []string{}
+	var ids []string
 	for _, h := range hosts {
 		ids = append(ids, h.Id)
 	}
+	if len(ids) == 0 {
+		return nil
+	}
 
-	if err := db.RemoveAll(evergreen.CredentialsCollection, bson.M{CertUserIDKey: bson.M{"$in": ids}}); err != nil {
-		return errors.Wrap(err, "could not delete credentials")
+	if _, err := db.UpdateAll(Collection, bson.M{
```

Comment:
Why not just do an `UpdateAll` with the original query? Is it just for the event logging below?

Reply:
It's for the event logging.